genesyscloud/util/util_retries.go

```diff
@@ -196,3 +203,80 @@ func IsTimeoutError(errDiag error) bool {
 	return strings.Contains(errStringLower, "timeout") ||
 		strings.Contains(errStringLower, "context deadline exceeded")
 }
+
+// GetRetryAfterDelay extracts and parses the Retry-After header from an APIResponse.
+// It returns the delay duration and true if the header was present and valid, false otherwise.
+// Retry-After can be either:
+//   - A number of seconds (integer)
+//   - An HTTP date (RFC 1123 format)
+func GetRetryAfterDelay(apiResponse *platformclientv2.APIResponse) (time.Duration, bool) {
+	if apiResponse == nil || apiResponse.Response == nil {
+		return 0, false
+	}
+
+	retryAfterHeader := apiResponse.Response.Header.Get("Retry-After")
+	if retryAfterHeader == "" {
+		return 0, false
+	}
+
+	// Try parsing as integer (seconds)
+	if seconds, err := strconv.Atoi(retryAfterHeader); err == nil {
+		if seconds > 0 {
+			return time.Duration(seconds) * time.Second, true
+		}
+		return 0, false
+	}
+
+	// Try parsing as HTTP date (RFC 1123 format: Mon, 02 Jan 2006 15:04:05 GMT)
+	if httpDate, err := time.Parse(time.RFC1123, retryAfterHeader); err == nil {
+		now := time.Now()
+		delay := httpDate.Sub(now)
+		if delay > 0 {
+			return delay, true
+		}
+		return 0, false
+	}
+
+	// Try parsing as RFC 1123 with timezone
+	if httpDate, err := time.Parse(time.RFC1123Z, retryAfterHeader); err == nil {
+		now := time.Now()
+		delay := httpDate.Sub(now)
+		if delay > 0 {
+			return delay, true
+		}
+		return 0, false
+	}
+
+	// Try parsing as RFC 822 format (alternative date format)
+	if httpDate, err := time.Parse(time.RFC822, retryAfterHeader); err == nil {
```

Comment:
There seems to be a lot of duplication here. The code seems identical except for the format. Could you not just make a loop that iterates through the 3 formats.

Reply:
I also didn't even realize that all these formats were valid? Is that more an HTTP spec thing and not a genesys platform thing? Because all of our public api doc only lists retry-after as seconds as the format, right? I mean RTA would probably completely ignore Retry-After if you called WFM service and I gave you a date timestamp back, right?

Reply:
Still waiting on answer to this. To me the top choices are aligning with platform standards or fully aligning with the spec. Although I still feel like all of this is unnecessary. This is only used on genesys APIs, right? If any API started returning Retry-After as something than an integer, it would seem like that should be viewed as a significant breaking change.